src/oss/langchain/multi-agent.mdx

```diff
@@ -7,85 +7,85 @@ Instead of relying on a single agent to handle every step, **multi-agent archite
-| [**Tool Calling**](#tool-calling) | A **supervisor** agent calls other agents as *tools*. The “tool” agents don’t talk to the user directly — they just run their task and return results.                  | Centralized: all routing passes through the calling agent. | Task orchestration, structured workflows.        |
-| [**Handoffs**](#handoffs)         | The current agent decides to **transfer control** to another agent. The active agent changes, and the user may continue interacting directly with the new agent. | Decentralized: agents can change who is active.            | Multi-domain conversations, specialist takeover. |
+At the center of multi-agent design is <Tooltip tip="The practice of carefully designing what information (prompts, history, state) is available to an LLM at each step">**[context engineering](/oss/langchain/context-engineering)**</Tooltip> — deciding what information each agent sees. LangChain gives you fine-grained control over:
 
-<Card
-    title="Tutorial: Build a supervisor agent"
-    icon="sitemap"
-    href="/oss/langchain/supervisor"
-    arrow cta="Learn more"
->
-    Learn how to build a personal assistant using the supervisor pattern, where a central supervisor agent coordinates specialized worker agents.
-    This tutorial demonstrates:
-    * Creating specialized sub-agents for different domains (calendar and email)
-    * Wrapping sub-agents as tools for centralized orchestration
-    * Adding human-in-the-loop review for sensitive actions
-</Card>
+* Which parts of the conversation or state are passed to each agent.
+* Specialized prompts tailored to sub-agents.
+* Inclusion/exclusion of intermediate reasoning.
+* Customizing input/output formats per agent.
 
+The quality of your system depends on context engineering. The goal is to ensure that each agent has access to the correct data it needs to perform its task, whether it's acting as a tool or as an active agent.
 
-## Choosing a pattern
+## Multi-agent patterns
 
-| Question                                              | Tool Calling | Handoffs |
-|-------------------------------------------------------|--------------|----------|
-| Need centralized control over workflow?               | ✅ Yes        | ❌ No     |
-| Want agents to interact directly with the user?       | ❌ No         | ✅ Yes    |
-| Complex, human-like conversation between specialists? | ❌ Limited    | ✅ Strong |
+There are three main multi-agent patterns:
+
+| Pattern | How it works | Example use case |
+|---------|--------------|------------------|
+| [**Tool calling**](#tool-calling) | A supervisor agent calls other agents as tools. **Centralized** control — all routing passes through the supervisor. | Task orchestration, parallel execution |
+| [**Handoffs**](#handoffs) | Agents transfer control to each other via tool calls or graph edges. **Decentralized** — the active agent changes dynamically. | Customer support, multi-domain conversations |
```

Comment:
The classifier pattern that runs on every user input is very common and not captured in this classification